35420082 openLDAP libldap does not specify AI_ADDRCONFIG by default causing IPv6 DNS AAAA lookups

Solaris specific patch to use AI_ADDRCONFIG when calling getaddrinfo()
and to not hold lock as getaddrinfo(3C) is MT-safe on Solaris.

--- openldap-2.4.59/libraries/libldap/os-ip.c
+++ openldap-2.4.59/libraries/libldap/os-ip.c
@@ -594,7 +594,7 @@ }
 
 #if defined( HAVE_GETADDRINFO ) && defined( HAVE_INET_NTOP )
 	memset( &hints, '\0', sizeof(hints) );
-#ifdef USE_AI_ADDRCONFIG /* FIXME: configure test needed */
+#if defined(USE_AI_ADDRCONFIG) || defined(__sun) /* FIXME: configure test needed */
 	/* Use AI_ADDRCONFIG only on systems where its known to be needed. */
 	hints.ai_flags = AI_ADDRCONFIG;
 #endif
@@ -602,12 +602,16 @@ hints.ai_family = ldap_int_inet4or6;
 	hints.ai_socktype = socktype;
 	snprintf(serv, sizeof serv, "%d", port );
 
+#if !defined(__sun)
 	/* most getaddrinfo(3) use non-threadsafe resolver libraries */
 	LDAP_MUTEX_LOCK(&ldap_int_resolv_mutex);
+#endif
 
 	err = getaddrinfo( host, serv, &hints, &res );
 
+#if !defined(__sun)
 	LDAP_MUTEX_UNLOCK(&ldap_int_resolv_mutex);
+#endif
 
 	if ( err != 0 ) {
 		osip_debug(ld, "ldap_connect_to_host: getaddrinfo failed: %s\n",
